test/unit/vdom/properties.js

```diff
@@ -117,4 +117,63 @@ describe('vdom/properties', () => {
       );
     });
   });
+
+  describe('#876 - Apply properties and attributes in a polyfill environment', () => {
+    it('setting properties (original issue)', (done) => {
+      const Elem1 = define('x-test', {
+        render (elem) {
+          return h(Elem2, { fooBar: true });
+        }
+      });
+      const Elem2 = define('x-test', {
+        props: {
+          fooBar: {}
+        },
+        render(elem) {
+          expect(elem.fooBar).to.be.equal(true);
+          done();
+        }
+      });
+      const elem = new Elem1();
+      fixture(elem);
+    });
+
+    describe('setting attributes', () => {
+      it(`observedAttributes: ['fooBar']`, (done) => {
+        const Elem1 = define('x-test', {
+          render (elem) {
+            return h(Elem2, { fooBar: true });
+          }
+        });
+        const Elem2 = define('x-test', {
+          observedAttributes: ['fooBar'],
+          render(elem) {
+            expect(elem.fooBar).to.be.equal(undefined);
+            expect(elem.getAttribute('foobar')).to.equal('true');
```

Comment:
I ended up refactoring the tests into a single one because we couldn't used `observedAttributes`.